logic/src/commonMain/kotlin/com/wire/kalium/logic/util/ServerTimeHandler.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.util
+
+import kotlinx.datetime.Clock
+
+object ServerTimeHandler {
+    private var timeOffset: Long = 0
```

Comment:
Question: Would this mutable var from the singleton, handle correctly the timezone changes, ie: daylight saving or traveling ? (in case we don't restart the app) food for thought

I have the feeling since we are recalculating every time we receive an event is not big of a deal. But mutable vars on a singletons, sometimes can go wrong

Reply:
Yes and that's the point of recalculating frequently the timeOffset, to not become outdated and handle device's timezone changes.

Reply:
Super silly thing, maybe, but:
I wonder what if there are two accounts on the same device, and they use two different servers, and there's a mismatch between these servers 😅
Maybe it would be best if it wasn't a singleton, but rather one instance per user.